ebill_postfinance/models/ebill_postfinance_service.py

```diff
@@ -0,0 +1,153 @@
+            ("XML", "XML Yellow Bill"),
+            ("EAI.XML", "Custom XML (SAPiDoc)"),
+            # ("eai.edi", "Custom EDIFACT"),
+            ("struct.pdf", "Factur X"),
+        ],
+    )
+    use_file_type_xml_paynet = fields.Boolean(
+        string="Use Paynet/SIX format",
+        help="Enable use of legacy SIX/Paynet invoice format.",
+    )
+    operation_timeout = fields.Integer(
+        string="HTTP Timeout",
+        default="600",
+        help="Timeout for each HTTP (GET, POST) request in seconds.",
+    )
+
+    def _get_service(self):
+        return ebilling_postfinance.WebService(
+            self.use_test_service,
+            self.username,
+            self.password,
+            self.biller_id,
+            self.operation_timeout,
+        )
+
+    def test_ping(self):
+        """Test the service from the UI."""
+        self.ensure_one()
+        msg = ["Test connection to service"]
+        res = self.ping_service()
+        if res:
+            msg.append("Success pinging service \n  Receive :{}".format(res))
+        else:
+            msg.append(" - Failed pinging service")
+        raise UserError("\n".join(msg))
+
+    def ping_service(self, test_error=False, test_exception=False):
+        """Ping the service, uses the authentication.
+
+        test_error: will create an unhandled error in the repsonse
+        test_exception: will create a FaultException
+
+        """
+        service = self._get_service()
+        return service.ping()
+
+    def search_invoice(self, transaction_id=None):
+        """Get invoice status from the server.
+
+        transaction_id:
+        """
+        service = self._get_service()
+        res = service.search_invoices(transaction_id)
+        if res.InvoiceCount == 0:
+            _logger.info("Search invoice returned no invoice")
+            return res
+        if res.InvoiceCount < res.TotalInvoiceCount:
+            # TODO handle the case where there is more to download ?
+            _logger.info("Search invoice has more to download")
+        for message in res.InvoiceList.SearchInvoice:
+            _logger.info(f"Found record for message {message}")
+            record = self.invoice_message_ids.search(
+                [("transaction_id", "=", message.TransactionId)]
+            )
+            if record:
+                record.update_message_from_server_data(message)
+            else:
+                _logger.warning(f"Could not find record for message {message}")
+        return res
+
+    def upload_file(self, transaction_id, file_type, data):
+        service = self._get_service()
+        res = service.upload_files(transaction_id, file_type, data)
+        return res
+
+    def get_invoice_list(self, archive_data=False):
+        service = self._get_service()
+        res = service.get_invoice_list(archive_data)
+        return res
+
+    def get_process_protocol_list(self, archive_data=False):
+        # Is this the processing result of an invoice ?
```

Comment:
Should it translate as a TODO ?

Reply:
Questioning the use of this call to the api.
The situation at the moment is, most possible call to the api are implemented, so they can be used and tested.
But they are not all used yet by the module for real.
There is the roadmap file that contains some more information.